### PR TITLE
Add New Feature and Refactor Proxy Server

### PR DESCRIPTION
**Pull request by evan-m-jackson**
- Add Port configuration
- Refactor YAML to deploy only on pushes to main
- Refactor YAML to run tests and prevent merge if any tests fail

### CODE REVIEW

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        var proxyStream = new List<string>() { "HTTP/1.1 200 OK", "Content-Type: application/json", "Content-Length: 20", "", "{ 'task': 'First Item' }" };
+        var stream = new List<string>();
+        var reader = new TestReadStreams(proxyStream);
+        var httpType = "POST";
+        var httpPath = "todo";
+        var writer = new TestWriteStreams(stream);
+
+        var proxyResponse = new ProxyResponse(reader, writer, httpPath, httpType);
+        proxyResponse.GetResponse();
+		var actualStatusLine = stream[0];
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Got an indentation issue here.

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+namespace HTTPServerProxyTests.Response;
+
+public class UnitTestsForProxyResponse
+{
+    [Fact]
+    public void GetTodoPOSTValidResponseTest()
+    {
+        var expectedStatusLine = "HTTP/1.1 201 Created";
+        var proxyStream = new List<string>() { "HTTP/1.1 200 OK", "Content-Type: application/json", "Content-Length: 20", "", "{ 'task': 'First Item' }" };
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: These lists should broken up into multiple lines - they are really long.

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs [hunk 3]
```diff
@@ -0,0 +1,178 @@
+using HTTPServerReadTests.Streams;
+using HTTPServerProxy.Response;
+using HTTPServerWriteTests.Streams;
+
+namespace HTTPServerProxyTests.Response;
+
+public class UnitTestsForProxyResponse
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Nitpick here but the name should probably ProxyResponseTest or TestProxyResposne - so I can tell what class is being tested at a glance. And "UnitTestsFor" is a little long and unconventional.

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    {
+        var expectedStatusLine = "HTTP/1.1 201 Created";
+        var proxyStream = new List<string>() { "HTTP/1.1 200 OK", "Content-Type: application/json", "Content-Length: 20", "", "{ 'task': 'First Item' }" };
+        var stream = new List<string>();
+        var reader = new TestReadStreams(proxyStream);
+        var httpType = "POST";
+        var httpPath = "todo";
+        var writer = new TestWriteStreams(stream);
+
+        var proxyResponse = new ProxyResponse(reader, writer, httpPath, httpType);
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Giving the number of constructor parameters here you might benefit from a Factory function or the Builder pattern. A less complex solution would be to use named parameters here. CleanCode uses a guideline of 2 parameters or less to any function, unless you can name them.

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        proxyResponse.GetResponse();
+		var actualStatusLine = stream[0];
+        Assert.Equal(expectedStatusLine, actualStatusLine);
+    }
+
+	[Fact]
+    public void GetTodoPOSTValidResponseNoBodyTest()
```
Comment (paytonrules) [inline HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:34]: Good name (the NoBody made it obvious for me what you were testing).

File: HTTPServerProject/Program.cs [hunk 6]
```diff
@@ -40,34 +48,53 @@ public void Conversation()
 
             var httpType = header.GetRequestType(initialLine);
             var httpPath = header.GetPath(initialLine);
+
```
Comment (paytonrules) [inline HTTPServerProject/Program.cs:51]: Got some indentation issues here.

**Review by paytonrules — COMMENTED** [review]
Not done - but I realized I was focusing too much on micro and want to step back and look at this at a higher level, so let me get these comments in now.

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs (hunk 5 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:34]: Thanks!

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs (hunk 1 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Agreed. Will update

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs (hunk 2 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Agreed. Will update

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs (hunk 3 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Agreed. Will update

File: HTTPServerProject/Program.cs (hunk 6 above)
Comment (evan-m-jackson) [reply HTTPServerProject/Program.cs:51]: Agreed. Will update

File: HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs (hunk 4 above)
Comment (evan-m-jackson) [reply HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs]: Sounds good. I think I'll just use named parameters

**Pull request merged**